hocon/value.go

```diff
@@ -242,202 +287,323 @@ func (p *HoconValue) NewValue(value HoconElement) {
 		}
 	}
 
-	return arrs
+	return items, nil
 }
 
-func (p *HoconValue) GetChildObject(key string) *HoconValue {
-	obj := p.GetObject()
-	if obj == nil {
-		return nil
+func (p *HoconValue) GetChildObject(key string) (*HoconValue, error) {
+	objectV, err := p.GetObject()
+	if err != nil {
+		return nil, err
 	}
-	return obj.GetKey(key)
+
+	if objectV == nil {
+		return nil, nil
+	}
+
+	return objectV.GetKey(key), nil
 }
 
 func (p *HoconValue) IsArray() bool {
-	return p.GetArray() != nil
+	arr, err := p.GetArray()
+	if err != nil {
+		return false
+	}
+
+	return arr != nil
 }
 
-func (p *HoconValue) GetTimeDuration(allowInfinite bool) time.Duration {
-	res := p.GetString()
-	groups, matched := findStringSubmatchMap(res, `^(?P<value>([0-9]+(\.[0-9]+)?))\s*(?P<unit>(nanoseconds|nanosecond|nanos|nano|ns|microseconds|microsecond|micros|micro|us|milliseconds|millisecond|millis|milli|ms|seconds|second|s|minutes|minute|m|hours|hour|h|days|day|d))$`)
+func (p *HoconValue) GetTimeDuration(allowInfinite bool) (time.Duration, error) {
+	stringV, err := p.GetString()
+	if err != nil {
+		return 0, err
+	}
+
+	groups, matched := findStringSubmatchMap(stringV, `^(?P<value>([0-9]+(\.[0-9]+)?))\s*(?P<unit>(nanoseconds|nanosecond|nanos|nano|ns|microseconds|microsecond|micros|micro|us|milliseconds|millisecond|millis|milli|ms|seconds|second|s|minutes|minute|m|hours|hour|h|days|day|d))$`)
 
 	if matched {
-		u := groups["unit"]
-		strV := groups["value"]
-		v := parsePositiveValue(strV)
+		foundUnit, foundFloat := groups["unit"], groups["value"]
+		floatV, err := parsePositiveValue(foundFloat)
+		if err != nil {
+			return 0, err
+		}
 
-		switch u {
+		switch foundUnit {
 		case "nanoseconds", "nanosecond", "nanos", "nano", "ns":
-			return time.Duration(float64(time.Nanosecond) * v)
+			return time.Duration(float64(time.Nanosecond) * floatV), nil
 		case "microseconds", "microsecond", "micros", "micro":
-			return time.Duration(float64(time.Microsecond) * v)
+			return time.Duration(float64(time.Microsecond) * floatV), nil
 		case "milliseconds", "millisecond", "millis", "milli", "ms":
-			return time.Duration(float64(time.Millisecond) * v)
+			return time.Duration(float64(time.Millisecond) * floatV), nil
 		case "seconds", "second", "s":
-			return time.Duration(float64(time.Second) * v)
+			return time.Duration(float64(time.Second) * floatV), nil
 		case "minutes", "minute", "m":
-			return time.Duration(float64(time.Minute) * v)
+			return time.Duration(float64(time.Minute) * floatV), nil
 		case "hours", "hour", "h":
-			return time.Duration(float64(time.Hour) * v)
+			return time.Duration(float64(time.Hour) * floatV), nil
 		case "days", "day", "d":
-			return time.Duration(float64(time.Hour*24) * v)
+			return time.Duration(float64(time.Hour*24) * floatV), nil
 		}
+
+		return 0, fmt.Errorf("cannot parse time value: %s", stringV)
 	}
 
-	if strings.ToLower(res) == "infinite" {
+	if strings.ToLower(stringV) == "infinite" {
```

Comment:
"infinite" should be constant in special file with constants or at header of this file

Reply:
Done